raiden-cli/src/routes/channels.ts

```diff
@@ -52,7 +64,91 @@ async function openChannel(this: Cli, request: Request, response: Response, next
   } catch (error) {
     if (isInvalidParameterError(error)) {
       response.status(400).send(error.message);
-    } else if (error.code === 'INSUFFICIENT_FUNDS') {
+    } else if (isInvalidParameterError(error)) {
+      response.status(402).send(error.message);
+    } else if (isConflictError(error)) {
+      response.status(409).send(error.message);
+    } else {
+      next(error);
+    }
+  }
+}
+
+const allowedUpdateKeys = new Set(['state', 'total_deposit', 'total_withdraw']);
+function validateChannelUpdateBody(request: Request, response: Response, next: NextFunction) {
+  const intersec = Object.keys(request.body).filter((k) => allowedUpdateKeys.has(k));
+  if (intersec.length < 1)
+    return response
+      .status(400)
+      .send('one of "state" | "total_deposit" | "total_withdraw" operations required');
+  else if (intersec.length > 1)
+    return response
+      .status(409)
+      .send('more than one of "state" | "total_deposit" | "total_withdraw" requested');
+  if (
+    request.body.state &&
+    ![ApiChannelState.closed, ApiChannelState.settled].includes(request.body.state)
+  )
+    return response
+      .status(400)
+      .send('invalid "state" requested: must be one of "closed" | "settled"');
```

Comment:
Prone to errors. Simply use the enumeration values directly. No need to differ/replicate between the check and the related message. Even it might be longer code for the message it is more correct. To be even more correct and avoid any kind of mistakes on future changes I would put the accepted states into a list as you did for the keys above. Then you can declare it once and use it twice, making sure these both lists will never deviate from each other.

I know this are nitpicks, but I think this brings the code to another level of quality. Most of the time you can say it is super unlikely that this happens and is not necessary. I think code should attempt to be as self-contained as possible, deleting the instable human factor. Also does it come with some piece in mind. You can code free without worrying to break anything. This might be not an actual break, but many of these small mistakes lead to bad software over time.